### PR TITLE
Refactor tests around NPQ

### PR DESCRIPTION
**Pull request by asmega**
### Context

- There is a callback `NPQValidationData#update_participant_profile`
- It creates and persists multiple database records
- It needs to be removed so we can have finer grain control of when it happens and the behaviour of it
- However many tests rely on this mechanism for the setup of its test data
- Therefore removing this method cannot be done first time round
- By reducing the dependency of this mechanism in the tests and instead setting up the data correctly this callback can be more easily removed in the next change

### Changes proposed in this pull request

- Removes most of test reliance on `NPQValidationData#update_participant_profile`

### Guidance to review

- There should be no behavioural change
- The tests should continue to work if `NPQValidationData#update_participant_profile` had been deleted

### Testing

- n/a

### Review Checks
- [x] All pages have automated accessibility checks via cypress
- [x] All pages have visual tests via cypress + percy
- [x] All `School` queries are correctly scoped - `eligible` when they need to be

### How can I view this in a review app?

- No behavioural changes have been made so app should behave the same

### CODE REVIEW

File: spec/services/record_participant_declaration_service_spec.rb [hunk 1]
```diff
@@ -5,85 +5,119 @@
...
       it "raises a ParameterMissing error" do
         expect { described_class.call(params) }.to raise_error(ActionController::ParameterMissing)
       end
     end
 
     context "when valid user is an early_career_teacher" do
+      before do
+        school_cohort
+        partnership
+      end
```
Comment (kicferk1) [inline spec/services/record_participant_declaration_service_spec.rb:79]: Can we use `let!` where we define them to ensure they are run before tests?

File: spec/services/record_participant_declaration_service_spec.rb [hunk 2]
```diff
@@ -5,85 +5,119 @@
 RSpec.describe RecordParticipantDeclaration do
   let(:ecf_lead_provider) { create(:lead_provider) }
   let(:cpd_lead_provider) { create(:cpd_lead_provider, lead_provider: ecf_lead_provider) }
-  let(:another_lead_provider) { create(:cpd_lead_provider, name: "Unknown") }
+  let(:another_cpd_lead_provider) { create(:cpd_lead_provider, name: "Unknown") }
```
Comment (kicferk1) [inline spec/services/record_participant_declaration_service_spec.rb:8]: I smell... conflicts.

In here and in shared setup, with the work I've done on https://github.com/DFE-Digital/early-careers-framework/pull/859/files#diff-b04de0b996bbd7c765305b53215f91f9edad6ef4df41f5fe83615d74a3ffb898L5

I'm happy to deal with the shared setup conflicts myself, but could you use the diff from the link above to make this test use common track and pay test setup? When you do, I hope things will 'just work' - and if not I am happy to pair on it.

Comment (asmega) [reply spec/services/record_participant_declaration_service_spec.rb:8]: I think it's just easier to fix the conflicts after the merge. Feel free to merge first and I'll patch up afterwards

File: spec/services/record_participant_declaration_service_spec.rb (hunk 1 above)
Comment (asmega) [reply spec/services/record_participant_declaration_service_spec.rb:79]: I can't replace the `let` with `let!`. The test will fail as the behaviour and outcome would be different. The tests are written in a quirky way. Keeping them lazy then evaluating when needed seems to me the best solution. Please let me know if you have a better suggestion

File: spec/services/record_participant_declaration_service_spec.rb (hunk 2 above)
Comment (asmega) [reply spec/services/record_participant_declaration_service_spec.rb:8]: oh wait. i misunderstood and was reading another part of the PR. hmm do we really want these as shared contexts? i foresee these being painful sharing all this setup stuff. if someone wants to test something slightly different it just won't work as it's fully coupled. I'd rather opt for duplication here personally.

i see shared contexts more of a way to test for a very specific behaviour common to many things. are we even in that area?

Comment (kicferk1) [reply spec/services/record_participant_declaration_service_spec.rb:8]: I think we want a base shared setup for T&P tests - and to diverge from that setup if some individual tests require it. I think almost all T&P tests require some ECF participant in a school which is in a partnership with a lead provider, and writing it down everywhere is a pain. This is simply a place where we haven't used the shared setup yet (it is a pretty new thing).

**Review by ethanmills — APPROVED** [review]

**Comment by asmega** [comment]
Other work has affected this PR so I'll open a new one up in a similar vein when I'm done

**Pull request closed without merging**